Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -913,7 +913,7 @@ final class DeclarationTests: ParserTestCase {
       }
       """,
       diagnostics: [
-        DiagnosticSpec(message: "expected '->' and return type in subscript", fixIts: ["insert '->' and return type"])
+        DiagnosticSpec(message: "expected '->' and return type in return clause", fixIts: ["insert '->' and return type"])
```

Comment:
I think I like the old diagnostic better and would thus prefer `returnClause` to not have a `nameForDiagnostic`. “return type in return clause” just sounds weird.